src/CodeGen_Xtensa.cpp

```diff
@@ -77,9 +79,10 @@ extern "C" {
 
 extern void *halide_tcm_malloc(void *user_context, size_t x);
 extern void halide_tcm_free(void *user_context, void *ptr);
-extern int halide_init_dma();
-extern int32_t halide_xtensa_copy_1d(void* dst, int32_t dst_base, void* src, int32_t src_base, int extent, int item_size);
-extern int32_t halide_xtensa_wait_for_copy(int32_t id);
+extern int halide_init_dma(int channel_count);
+extern int32_t halide_xtensa_copy_1d(int32_t channel, void* dst, int32_t dst_base, void* src, int32_t src_base, int extent, int item_size);
```

Comment:
nit: it's a little weird to mix `int` and `int32_t` here; I'm sure it's fine, it's just that reading it made me wonder if the difference was necessary or just arbitrary.

Reply:
Fixed, changed them all into int32_t.